File: task/src/main/java/com/acutecoder/task/TaskException.java
package com.acutecoder.task;

public class TaskException extends RuntimeException {
    public TaskException(Exception e, boolean isBackground) {
        super("Error while executing " + (isBackground ? "Background" : "Foreground") + " task", e);
    }

    public TaskException(String msg, boolean isBackground) {
        super("Error while executing " + (isBackground ? "Background" : "Foreground") + " task: " + msg);
    }
}

File: task_base/src/main/java/com/acutecoder/task/TaskException.kt
package com.acutecoder.task

/**
 * Created by Bhuvaneshwaran
 *
 * 10:44 PM, 9/3/2023
 * @author AcuteCoder
 */

class TaskException : RuntimeException {
 constructor(
  e: Exception?, isBackground: Boolean
 ) : super(
 "Error while executing " + (if (isBackground) "Background" else "Foreground") + " task", e
 )

 constructor(
  msg: String, isBackground: Boolean
 ) : super("Error while executing " + (if (isBackground) "Background" else "Foreground") + " task: " + msg)
}